src/shared/filters/api-exception.filter.ts

```diff
@@ -0,0 +1,310 @@
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "The request was denied by the server. Please check the request",
+          error: exception.message,
+          timestamp: new Date().toISOString(),
+          path: httpAdapter.getRequestUrl(ctx.getRequest()),
+        };
+
+        console.log("Response to be sent :: " + responseBody)
+    
+        httpAdapter.reply(ctx.getResponse(), responseBody, exception.getStatus());
+    }
+}
+
+@Catch(RequestTimeoutException)
+class RequestTimeoutExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Time-out. The server took too long to respond. Please try again later.",
+          error: exception.message,
+          timestamp: new Date().toISOString(),
+          path: httpAdapter.getRequestUrl(ctx.getRequest()),
+        };
+
+        console.log("Response to be sent :: " + responseBody)
+    
+        httpAdapter.reply(ctx.getResponse(), responseBody, exception.getStatus());
+    }
+}
+
+@Catch(ConflictException)
+class ConflictExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Conflict - Your request could not be processed.",
+          error: exception.message,
+          timestamp: new Date().toISOString(),
+          path: httpAdapter.getRequestUrl(ctx.getRequest()),
+        };
+
+        console.log("Response to be sent :: " + responseBody)
+    
+        httpAdapter.reply(ctx.getResponse(), responseBody, exception.getStatus());
+    }
+}
+
+@Catch(GoneException)
+class GoneExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Your requested resource is no longer available and will not be available again.",
+          error: exception.message,
```

Comment:
The requested resource is no longer available.

Reply:
done